File: Java/src/tema9programacionorientadaaobjetos/excepciones/Ejercicio1.java
package tema9programacionorientadaaobjetos.excepciones;

import java.util.Scanner;

/**
 * <p>
 * Realiza un programa que pida 6 números por teclado y nos diga cuál es el
 * máximo. Si el usuario introduce un dato erróneo (algo que no sea un número
 * entero) el programa debe indicarlo y debe pedir de nuevo el número.
 * </p>
 * 
 * @author Francisco Javier Frías Serrano
 * @version 1.0
 */
public class Ejercicio1 {

	public static void main(String[] args) {

		Scanner entrada = new Scanner(System.in);

		System.out.println("Introduzca 6 números enteros...");

		int maximo = Integer.MIN_VALUE;

		for (int i = 1; i < 7; i++) {

			int numero = 0;
			boolean entradaValida = false;

			do {

				try {
					System.out.print("Nº " + i + ": ");
					numero = entrada.nextInt();
					entradaValida = true;

				} catch (Exception e) {
					System.err.println("Dato introducido inválido. Debe introducir un número entero.");
					System.err.println("Vuelva a introducir el dato");
					entrada.nextLine();
				}

			} while (!entradaValida);

			if (numero > maximo) {
				maximo = numero;
			}
		}

		System.out.println("El número máximo es " + maximo);
		entrada.close();
	}
}

File: Python/poo/excepciones/Ejercicio1.py
'''
Realiza un programa que pida 6 números por teclado y nos diga cuál es el máximo. Si el usuario introduce un dato erróneo 
(algo que no sea un número entero) el programa debe indicarlo y debe pedir de nuevo el número.

@author Francisco Javier Frías Serrano
@version 1.0
'''
print("Introduzca 6 números enteros...");

for i in range(1, 7):
            
    numero = 0 
    datoValido = False
    maximo = numero
            
    while not datoValido:

        try :
            numero = int(input("Nº " + str(i) + ": "))
            datoValido = True
            
            if (numero > maximo):
              maximo = numero
              
        except:
            print("Dato introducido inválido. Debe introducir un número entero.")
            print("Vuelva a introducir el dato")
            datoValido = False

    if (numero > maximo):
        maximo = numero

print("El número máximo es " , maximo);
